src/ripple/overlay/impl/PeerImp.h

```diff
@@ -198,6 +199,26 @@ class PeerImp
 
     friend class OverlayImpl;
 
+    class Metrics {
+    public:
+        void add_message(std::uint64_t bytes);
+        std::uint64_t average_bytes() const;
+        std::uint64_t total_bytes() const;
+
+    private:
+        std::uint64_t totalBytes = 0;
+        std::uint64_t accumBytes = 0;
+        std::uint64_t rollingAvgBytes = 0;
+        std::array<std::uint64_t, 30> rollingAvg = {};
```

Comment:
Didn't know that existed (I should just assume `boost` has everything), but agreed, I'll take a look at it.

Reply:
I believe `boost::circular_buffer` is a heap-based container.  Since `Metrics` is not being moved, `std::array` as an advantage in being stack allocated.

Reply:
I opt'd for readability this time around...

Reply:
It makes sense that it's heap-based, but I expected a fixed-sized version exists. I think that's OK for our purposes, and simplifies things, so I'm ok with it. Happy to reconsider if you have strong objections, @HowardHinnant.

Reply:
No objections here.